### ISSUE TITLE
Cannot run GPU on tensorflow Docker Image with Tensorflow 2.3.1

### ISSUE THREAD

**Issue by victormaricato** [tensorflow/tensorflow#45011]
**System information**
- OS Platform and Distribution: **[Tensorflow Docker Image](https://hub.docker.com/r/tensorflow/tensorflow/)**:
- TensorFlow installed from (source or binary): **`python3 -m pip install tensorflow-gpu`**
- TensorFlow version: **2.3.1**
- Python version: **3.7.5**
- Installed using virtualenv? pip? conda?: **pip (but it comes by default on docker image as far as I know)**
- GCC/Compiler version (if compiling from source): `7.5.0`
- CUDA/cuDNN version: ```nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243```
- GPU model and memory: **NVIDIA Tesla T4, 16GB. Created using `Compute Engine Deep Learning VM API` on Google Cloud Platform.**

After deploying a GCP Compute Engine Deep Learning VM, I am trying to use docker-compose to train a model using GPU.

Inside docker, I execute:
```
python3 -c "import tensorflow as tf; tf.test.is_gpu_available()"
```

Which in output (`False`), there is this line: (Full output available at the end)

`Could not load dynamic library 'libcublas.so.10'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64`

Something is happening that TensorFlow GPU 2.3.1 cannot lead this `libcublas.so.10` library.

Outside docker, this command returns `True`.

Here's my docker-compose and Dockerfile:

**docker-compose**, `version: 2.3`:
```
    analyses-gpu:
        container_name: analyses-gpu
        runtime: nvidia
        build:
            context: .
            dockerfile: .dockerfiles/DockerfileGPU
        working_dir: /app
        volumes:
            - ./:/app
        environment:
            - NVIDIA_VISIBLE_DEVICES=all
        command: "bash -c 'cd analyses/; make run'"
```

**Dockerfile**:
```
FROM tensorflow/tensorflow:latest-gpu

WORKDIR /app

RUN apt-get update && apt-get upgrade -y
RUN apt-get install -y git g++ && \
    apt-get install -y build-essential

# Update Python
RUN apt-get install -y python3.7 python3-dev
RUN update-alternatives --install /usr/local/bin/python python /usr/bin/python3.7 1
RUN update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.7 1
```

Full output mentioned:
```
2020-11-19 15:04:46.263566: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
WARNING:tensorflow:From <string>:1: is_gpu_available (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.config.list_physical_devices('GPU')` instead.
2020-11-19 15:04:47.399117: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2020-11-19 15:04:47.409370: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 2300000000 Hz
2020-11-19 15:04:47.411095: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x56755c0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-11-19 15:04:47.411130: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-11-19 15:04:47.414750: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2020-11-19 15:04:48.394486: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-11-19 15:04:48.395282: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x56e3690 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2020-11-19 15:04:48.395322: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5
2020-11-19 15:04:48.395600: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-11-19 15:04:48.396263: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5
coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.75GiB deviceMemoryBandwidth: 298.08GiB/s
2020-11-19 15:04:48.396307: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-11-19 15:04:48.396762: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcublas.so.10'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2020-11-19 15:04:48.398793: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2020-11-19 15:04:48.399183: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2020-11-19 15:04:48.401310: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2020-11-19 15:04:48.402616: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2020-11-19 15:04:48.407285: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2020-11-19 15:04:48.407319: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1753] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2020-11-19 15:04:48.407350: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-11-19 15:04:48.407369: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      0 
2020-11-19 15:04:48.407384: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1276] 0:   N 
False
```

**Comment by tejaswid** [comment]
I had to add "**capabilities=compute,utility**" to the **gpus** flag while running the image. I had the same issue for tensorflow-gpu==1.15 custom installed in a docker image. Just like you said, tensorflow-gpu was able to find libcuda outside a docker container, but unable to do so from inside the container.

change the following commad
`docker run --gpus all image_name bash`

to 
`docker run --gpus 'all,"capabilities=compute,utility"' image_name bash`

[This](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html) is the reference I followed.
